google/resource_container_cluster.go

```diff
@@ -447,19 +472,35 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 	}
 
 	if v, ok := d.GetOk("additional_zones"); ok {
+		if isRegionalCluster {
+			return fmt.Errorf("'additional_zones' is incompatible with regional clusters")
+		}
 		locationsList := v.(*schema.Set).List()
 		locations := []string{}
 		for _, v := range locationsList {
 			location := v.(string)
 			locations = append(locations, location)
 			if location == zoneName {
-				return fmt.Errorf("additional_zones should not contain the original 'zone'.")
+				return fmt.Errorf("additional_zones should not contain the original 'zone'")
 			}
 		}
 		locations = append(locations, zoneName)
 		cluster.Locations = locations
 	}
 
+	if isRegionalCluster {
+		locations := []string{}
+		locations = append(locations, d.Get("region").(string))
```

Comment:
I think the locations field is still supposed to be just a list of zones- this will likely still work without this entire block (and the `node_locations` field)

Reply:
I see that this is not a new field added to support regional clusters. So this is not relevant to this PR. Removing this. But we'll have to add this later. As `additional_zones` is getting deprecated.